cont3xt/integration.js

```diff
@@ -379,7 +379,18 @@ class Integration {
     return res.send({ success: true, integrations: results });
   }
 
-  static async runIntegrationsList (shared, query, itype, integrations, onFinish) {
+  static async runIntegrationsList (shared, indicator, parentQuery, integrations, onFinish) {
+    const { query, itype } = indicator;
+
+    // initial write (ensures dependable tracking of indicator tree)
+    shared.res.write(JSON.stringify({ purpose: 'link', indicator, parentQuery }));
+    shared.res.write(',\n');
+
+    // do not reissue integrations if they have been started by a matching query reached via different descendants
+    if (shared.queriedSet.has(query)) { return; }
```

Comment:
For future proof should these 2 lines be ``` `{$itype}-${query}` ```

Reply:
Yea, probably. I added this in a few places.